### ISSUE TITLE
Typescript cannot detect type definitions with pnpm

### ISSUE THREAD

**Issue by nawordar** [TypeStrong/fork-ts-checker-webpack-plugin#585]
## Current behavior

When the project is managed by pnpm, trying to build the project with webpack will result in multiple errors about lacking typings.

<details>
<summary>pnpm build</summary>

```
> fork-ts-checker-webpack-plugin-129-mre@1.0.0 build /home/nawordar/projects/fork-ts-checker-webpack-plugin-129-mre
> webpack

asset main.js 2.78 KiB [compared for emit] (name: main)
./src/index.ts 141 bytes [built] [code generated]
./src/hello.ts 175 bytes [built] [code generated]

ERROR in src/hello.spec.ts:3:1
TS2582: Cannot find name 'describe'. Do you need to install type definitions for a test runner? Try `npm i --save-dev @types/jest` or `npm i --save-dev @types/mocha`.
    1 | import { hello } from "./hello";
    2 |
  > 3 | describe("hello", () => {
      | ^^^^^^^^
    4 |   it("returns expected value", () => {
    5 |     const received = hello();
    6 |     expect(received).toEqual("Hello world");

ERROR in src/hello.spec.ts:4:3
TS2582: Cannot find name 'it'. Do you need to install type definitions for a test runner? Try `npm i --save-dev @types/jest` or `npm i --save-dev @types/mocha`.
    2 |
    3 | describe("hello", () => {
  > 4 |   it("returns expected value", () => {
      |   ^^
    5 |     const received = hello();
    6 |     expect(received).toEqual("Hello world");
    7 |   });

ERROR in src/hello.spec.ts:6:5
TS2304: Cannot find name 'expect'.
    4 |   it("returns expected value", () => {
    5 |     const received = hello();
  > 6 |     expect(received).toEqual("Hello world");
      |     ^^^^^^
    7 |   });
    8 | });
    9 |

ERROR in webpack.config.ts:4:3
TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node`.
    2 |
    3 | const mode =
  > 4 |   process.env.NODE_ENV === "production" ? "production" : "development";
      |   ^^^^^^^
    5 |
    6 | export default {
    7 |   mode,

ERROR in webpack.config.ts:8:12
TS2304: Cannot find name '__dirname'.
     6 | export default {
     7 |   mode,
  >  8 |   context: __dirname,
       |            ^^^^^^^^^
     9 |   entry: './src/index.ts',
    10 |   resolve: {
    11 |     extensions: [".ts", ".tsx", ".js"],

webpack 5.30.0 compiled with 5 errors in 1164 ms
 ERROR  Command failed with exit code 1.
```

</details>

## Expected behavior

Build the project without warnings, as with plain `tsc`.

## Steps to reproduce the issue

1. Clone the MRE repository
2. Run `pnpm install && pnpm build`

## Issue reproduction repository

https://github.com/nawordar/fork-ts-checker-webpack-plugin-585-mre

## Environment
 - **fork-ts-checker-webpack-plugin**: ^6.2.0
 - **typescript**: ^4.2.3
 - **eslint**: None
 - **webpack**: ^5.30.0
 - **os**: Arch Linux

**Comment by StreetStrider** [comment]
@nawordar this may be the pnpm issue. Do you have default pnpm hoisting rules in play?
https://pnpm.io/npmrc#public-hoist-pattern
The default public hoist pattern hoists `types` packages, so they become visible by tsc type resolution (which does not follow node resolution).

**Comment by nawordar** [comment]
Yes, I'm using default rules. It looks like a problem with handling symlinks, because after issuing these commands:
```
mv node_modules/@types/node{,-bak}
cp -r node_modules/.pnpm/@types+node@14.14.37/node_modules/@types/node node_modules/@types
```
I got rid of errors related to `@types/node`.

**Comment by piotr-oles** [comment]
Hi! :wave: Thanks for reporting this issue. Just to double-check - if you run `tsc --watch --noEmit`, there are no such errors, right?

**Comment by nawordar** [comment]
Yes, there are no issues when I run `tsc --watch --noEmit`.

**Comment by piotr-oles** [comment]
I found the issue, fixed in #625 :)
